yt_dlp/extractor/reddit.py

```diff
@@ -147,19 +114,42 @@ def add_thumbnail(src):
                 for resolution in resolutions:
                     add_thumbnail(resolution)
 
-        return {
-            '_type': 'url_transparent',
+        info = {
+            'id': video_id,
             'url': video_url,
             'title': data.get('title'),
             'thumbnails': thumbnails,
             'timestamp': float_or_none(data.get('created_utc')),
             'uploader': data.get('author'),
-            'duration': int_or_none(try_get(
-                data,
-                (lambda x: x['media']['reddit_video']['duration'],
-                 lambda x: x['secure_media']['reddit_video']['duration']))),
             'like_count': int_or_none(data.get('ups')),
             'dislike_count': int_or_none(data.get('downs')),
             'comment_count': int_or_none(data.get('num_comments')),
             'age_limit': age_limit,
         }
+
+        # Check if media is hosted on reddit:
+        reddit_video = try_get(data, lambda x: x['media']['reddit_video'])
+        if reddit_video:
+            playlist_urls = [
+                try_get(reddit_video, lambda x: unescapeHTML(x[y])) for y in ('dash_url', 'hls_url')
+            ]
+
+            dash_playlist_url = playlist_urls[0] or f'https://v.redd.it/{video_id}/DASHPlaylist.mpd'
+            hls_playlist_url = playlist_urls[1] or f'https://v.redd.it/{video_id}/HLSPlaylist.m3u8'
+
+            formats = self._extract_m3u8_formats(
+                hls_playlist_url, video_id, 'mp4',
+                entry_protocol='m3u8_native', m3u8_id='hls', fatal=False)
+            formats.extend(self._extract_mpd_formats(
+                dash_playlist_url, video_id, mpd_id='dash', fatal=False))
+            self._sort_formats(formats)
+
+            info.update({
+                'formats': formats,
+                'duration': int_or_none(reddit_video.get('duration')),
+            })
+
+        else:  # Not hosted on reddit, must continue extraction
+            info['_type'] = 'url_transparent'
+
+        return info
```

Comment:
something like this is more readable imo

```suggestion
            return {
                **info,
                'formats': formats,
                'duration': int_or_none(reddit_video.get('duration')),
            })

        return {
            **info,
            '_type': 'url_transparent',
            'url': video_url,
        }
```

Reply:
Agreed 👍